### ISSUE TITLE
Fix link errors

### ISSUE THREAD

**Issue by kenz-gelsoft** [kenz-gelsoft/llvm-project#4]
By stubbing out for now. Part of #1
```
~/src/llvm-project/build> ninja lldb
...
[10/12] Linking CXX shared library lib/liblldb.so.12.0.1
FAILED: lib/liblldb.so.12.0.1 
: && /bin/c++ -fPIC -fPIC -fvisibility-inlines-hidden -Werror=date-time -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-maybe-uninitialized -Wno-class-memaccess -Wno-redundant-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -fdiagnostics-color -Wno-deprecated-declarations -Wno-unknown-pragmas -Wno-strict-aliasing -Wno-deprecated-register -Wno-vla-extension -g  -Wl,-z,defs   -Wl,-rpath-link,build/./lib   -Wl,--version-script,"build/tools/lldb/source/API/liblldb.exports" -shared -Wl,-soname,liblldb.so.12 -o lib/liblldb.so.12.0.1 tools/lldb/source/API/CMakeFiles/liblldb.dir/SBAddress.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBAttachInfo.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBBlock.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBBreakpoint.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBBreakpointLocation.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBBreakpointName.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBBreakpointOptionCommon.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBBroadcaster.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBCommandInterpreter.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBCommandInterpreterRunOptions.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBCommandReturnObject.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBCommunication.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBCompileUnit.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBData.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBDebugger.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBDeclaration.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBEnvironment.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBError.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBEvent.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBExecutionContext.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBExpressionOptions.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBFileSpec.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBFile.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBFileSpecList.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBFrame.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBFunction.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBHostOS.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBInstruction.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBInstructionList.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBLanguageRuntime.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBLaunchInfo.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBLineEntry.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBListener.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBMemoryRegionInfo.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBMemoryRegionInfoList.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBModule.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBModuleSpec.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBPlatform.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBProcess.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBProcessInfo.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBQueue.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBQueueItem.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBReproducer.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBSection.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBSourceManager.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBStream.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBStringList.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBStructuredData.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBSymbol.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBSymbolContext.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBSymbolContextList.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBTarget.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBThread.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBThreadCollection.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBThreadPlan.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBTrace.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBTraceOptions.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBType.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBTypeCategory.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBTypeEnumMember.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBTypeFilter.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBTypeFormat.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBTypeNameSpecifier.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBTypeSummary.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBTypeSynthetic.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBValue.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBValueList.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBVariablesOptions.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBWatchpoint.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SBUnixSignals.cpp.o tools/lldb/source/API/CMakeFiles/liblldb.dir/SystemInitializerFull.cpp.o  lib/liblldbBase.a  lib/liblldbBreakpoint.a  lib/liblldbCore.a  lib/liblldbDataFormatters.a  lib/liblldbExpression.a  lib/liblldbHost.a  lib/liblldbInitialization.a  lib/liblldbInterpreter.a  lib/liblldbSymbol.a  lib/liblldbTarget.a  lib/liblldbUtility.a  lib/liblldbPluginABIAArch64.a  lib/liblldbPluginABIARM.a  lib/liblldbPluginABIHexagon.a  lib/liblldbPluginABIMips.a  lib/liblldbPluginABIPowerPC.a  lib/liblldbPluginABISystemZ.a  lib/liblldbPluginABIX86.a  lib/liblldbPluginArchitectureArm.a  lib/liblldbPluginArchitectureMips.a  lib/liblldbPluginArchitecturePPC64.a  lib/liblldbPluginDisassemblerLLVMC.a  lib/liblldbPluginDynamicLoaderDarwinKernel.a  lib/liblldbPluginDynamicLoaderMacOSXDYLD.a  lib/liblldbPluginDynamicLoaderPosixDYLD.a  lib/liblldbPluginDynamicLoaderStatic.a  lib/liblldbPluginDynamicLoaderHexagonDYLD.a  lib/liblldbPluginDynamicLoaderWindowsDYLD.a  lib/liblldbPluginDynamicLoaderWasmDYLD.a  lib/liblldbPluginInstructionARM.a  lib/liblldbPluginInstructionARM64.a  lib/liblldbPluginInstructionMIPS.a  lib/liblldbPluginInstructionMIPS64.a  lib/liblldbPluginInstructionPPC64.a  lib/liblldbPluginInstrumentationRuntimeASan.a  lib/liblldbPluginInstrumentationRuntimeMainThreadChecker.a  lib/liblldbPluginInstrumentationRuntimeTSan.a  lib/liblldbPluginInstrumentationRuntimeUBSan.a  lib/liblldbPluginJITLoaderGDB.a  lib/liblldbPluginCPlusPlusLanguage.a  lib/liblldbPluginObjCLanguage.a  lib/liblldbPluginObjCPlusPlusLanguage.a  lib/liblldbPluginCXXItaniumABI.a  lib/liblldbPluginAppleObjCRuntime.a  lib/liblldbPluginRenderScriptRuntime.a  lib/liblldbPluginMemoryHistoryASan.a  lib/liblldbPluginObjectContainerBSDArchive.a  lib/liblldbPluginObjectContainerMachOArchive.a  lib/liblldbPluginObjectFileBreakpad.a  lib/liblldbPluginObjectFileELF.a  lib/liblldbPluginObjectFileMachO.a  lib/liblldbPluginObjectFilePDB.a  lib/liblldbPluginObjectFilePECOFF.a  lib/liblldbPluginObjectFileJIT.a  lib/liblldbPluginObjectFileWasm.a  lib/liblldbPluginPlatformLinux.a  lib/liblldbPluginPlatformFreeBSD.a  lib/liblldbPluginPlatformNetBSD.a  lib/liblldbPluginPlatformOpenBSD.a  lib/liblldbPluginPlatformMacOSX.a  lib/liblldbPluginPlatformWindows.a  lib/liblldbPluginPlatformGDB.a  lib/liblldbPluginPlatformAndroid.a  lib/liblldbPluginProcessGDBRemote.a  lib/liblldbPluginProcessElfCore.a  lib/liblldbPluginProcessMachCore.a  lib/liblldbPluginProcessMinidump.a  lib/liblldbPluginScriptInterpreterNone.a  lib/liblldbPluginStructuredDataDarwinLog.a  lib/liblldbPluginSymbolFileBreakpad.a  lib/liblldbPluginSymbolFileDWARF.a  lib/liblldbPluginSymbolFileSymtab.a  lib/liblldbPluginSymbolFilePDB.a  lib/liblldbPluginSystemRuntimeMacOSX.a  lib/liblldbPluginSymbolVendorELF.a  lib/liblldbPluginSymbolVendorWasm.a  lib/liblldbPluginTypeSystemClang.a  lib/liblldbPluginUnwindAssemblyInstEmulation.a  lib/liblldbPluginUnwindAssemblyX86.a  lib/liblldbPluginPlatformLinux.a  lib/liblldbPluginPlatformGDB.a  lib/liblldbPluginDynamicLoaderDarwinKernel.a  lib/liblldbPluginPlatformMacOSX.a  lib/liblldbPluginPlatformPOSIX.a  lib/liblldbPluginDynamicLoaderMacOSXDYLD.a  lib/liblldbPluginObjectFileMachO.a  lib/liblldbPluginDynamicLoaderPosixDYLD.a  lib/liblldbPluginProcessElfCore.a  lib/liblldbPluginDynamicLoaderPosixDYLD.a  lib/liblldbPluginProcessElfCore.a  lib/liblldbPluginObjectFileBreakpad.a  lib/liblldbPluginObjectFileELF.a  lib/liblldbPluginObjectFileWasm.a  lib/liblldbBreakpoint.a  lib/liblldbCore.a  lib/liblldbDataFormatters.a  lib/liblldbExpression.a  lib/liblldbInterpreter.a  lib/liblldbSymbol.a  lib/liblldbTarget.a  lib/liblldbPluginCPlusPlusLanguage.a  lib/liblldbPluginObjCLanguage.a  lib/liblldbPluginAppleObjCRuntime.a  lib/liblldbPluginRenderScriptRuntime.a  lib/liblldbPluginObjectFilePDB.a  lib/liblldbPluginObjectFileJIT.a  lib/liblldbPluginSymbolFileDWARF.a  lib/liblldbPluginSymbolFilePDB.a  lib/liblldbPluginTypeSystemClang.a  lib/liblldbCommands.a  lib/liblldbPluginExpressionParserClang.a  lib/liblldbPluginProcessUtility.a  lib/liblldbPluginClangCommon.a  lib/liblldbPluginCPPRuntime.a  lib/liblldbPluginSymbolFileNativePDB.a  lib/liblldbPluginObjCRuntime.a  lib/liblldbBreakpoint.a  lib/liblldbCore.a  lib/liblldbDataFormatters.a  lib/liblldbExpression.a  lib/liblldbInterpreter.a  lib/liblldbSymbol.a  lib/liblldbTarget.a  lib/liblldbPluginCPlusPlusLanguage.a  lib/liblldbPluginObjCLanguage.a  lib/liblldbPluginAppleObjCRuntime.a  lib/liblldbPluginRenderScriptRuntime.a  lib/liblldbPluginObjectFilePDB.a  lib/liblldbPluginObjectFileJIT.a  lib/liblldbPluginSymbolFileDWARF.a  lib/liblldbPluginSymbolFilePDB.a  lib/liblldbPluginTypeSystemClang.a  lib/liblldbCommands.a  lib/liblldbPluginExpressionParserClang.a  lib/liblldbPluginProcessUtility.a  lib/liblldbPluginClangCommon.a  lib/liblldbPluginCPPRuntime.a  lib/liblldbPluginSymbolFileNativePDB.a  lib/liblldbPluginObjCRuntime.a  lib/liblldbBreakpoint.a  lib/liblldbCore.a  lib/liblldbDataFormatters.a  lib/liblldbExpression.a  lib/liblldbInterpreter.a  lib/liblldbSymbol.a  lib/liblldbTarget.a  lib/liblldbPluginCPlusPlusLanguage.a  lib/liblldbPluginObjCLanguage.a  lib/liblldbPluginAppleObjCRuntime.a  lib/liblldbPluginRenderScriptRuntime.a  lib/liblldbPluginObjectFilePDB.a  lib/liblldbPluginObjectFileJIT.a  lib/liblldbPluginSymbolFileDWARF.a  lib/liblldbPluginSymbolFilePDB.a  lib/liblldbPluginTypeSystemClang.a  lib/liblldbCommands.a  lib/liblldbPluginExpressionParserClang.a  lib/liblldbPluginProcessUtility.a  lib/liblldbPluginClangCommon.a  lib/liblldbPluginCPPRuntime.a  lib/liblldbPluginSymbolFileNativePDB.a  lib/liblldbPluginObjCRuntime.a  lib/liblldbBreakpoint.a  lib/liblldbCore.a  lib/liblldbDataFormatters.a  lib/liblldbExpression.a  lib/liblldbInterpreter.a  lib/liblldbSymbol.a  lib/liblldbTarget.a  lib/liblldbPluginCPlusPlusLanguage.a  lib/liblldbPluginObjCLanguage.a  lib/liblldbPluginAppleObjCRuntime.a  lib/liblldbPluginRenderScriptRuntime.a  lib/liblldbPluginObjectFilePDB.a  lib/liblldbPluginObjectFileJIT.a  lib/liblldbPluginSymbolFileDWARF.a  lib/liblldbPluginSymbolFilePDB.a  lib/liblldbPluginTypeSystemClang.a  lib/liblldbCommands.a  lib/liblldbPluginExpressionParserClang.a  lib/liblldbPluginProcessUtility.a  lib/liblldbPluginClangCommon.a  lib/liblldbPluginCPPRuntime.a  lib/liblldbPluginSymbolFileNativePDB.a  lib/liblldbPluginObjCRuntime.a  lib/liblldbBreakpoint.a  lib/liblldbCore.a  lib/liblldbDataFormatters.a  lib/liblldbExpression.a  lib/liblldbInterpreter.a  lib/liblldbSymbol.a  lib/liblldbTarget.a  lib/liblldbPluginCPlusPlusLanguage.a  lib/liblldbPluginObjCLanguage.a  lib/liblldbPluginAppleObjCRuntime.a  lib/liblldbPluginRenderScriptRuntime.a  lib/liblldbPluginObjectFilePDB.a  lib/liblldbPluginObjectFileJIT.a  lib/liblldbPluginSymbolFileDWARF.a  lib/liblldbPluginSymbolFilePDB.a  lib/liblldbPluginTypeSystemClang.a  lib/liblldbCommands.a  lib/liblldbPluginExpressionParserClang.a  lib/liblldbPluginProcessUtility.a  lib/liblldbPluginClangCommon.a  lib/liblldbPluginCPPRuntime.a  lib/liblldbPluginSymbolFileNativePDB.a  lib/liblldbPluginObjCRuntime.a  lib/liblldbBase.a  lib/liblldbHost.a  /packages/libxml2-2.10.4-2/.self/develop/lib/libxml2.so  lib/liblldbUtility.a  lib/libclang-cpp.so.12  lib/libLLVM-12.so && :
ld: lib/liblldbHost.a(Socket.cpp.o): in function `lldb_private::Socket::Read(void*, unsigned long&)':
lldb/source/Host/common/Socket.cpp:327: undefined reference to `recv'
ld: lib/liblldbHost.a(Socket.cpp.o): in function `lldb_private::Socket::GetOption(int, int, int&)':
lldb/source/Host/common/Socket.cpp:410: undefined reference to `getsockopt'
ld: lib/liblldbHost.a(Socket.cpp.o): in function `lldb_private::Socket::SetOption(int, int, int)':
lldb/source/Host/common/Socket.cpp:417: undefined reference to `setsockopt'
ld: lib/liblldbHost.a(Socket.cpp.o): in function `lldb_private::Socket::Send(void const*, unsigned long)':
lldb/source/Host/common/Socket.cpp:422: undefined reference to `send'
ld: lib/liblldbHost.a(Socket.cpp.o): in function `lldb_private::Socket::CreateSocket(int, int, int, bool, lldb_private::Status&)':
lldb/source/Host/common/Socket.cpp:442: undefined reference to `socket'
ld: lib/liblldbHost.a(Socket.cpp.o): in function `lldb_private::Socket::AcceptSocket(int, sockaddr*, unsigned int*, bool, lldb_private::Status&)':
lldb/source/Host/common/Socket.cpp:477: undefined reference to `accept'
ld: lib/liblldbHost.a(TCPSocket.cpp.o): in function `lldb_private::TCPSocket::GetLocalPortNumber() const':
lldb/source/Host/common/TCPSocket.cpp:86: undefined reference to `getsockname'
ld: lldb/source/Host/common/TCPSocket.cpp:91: undefined reference to `getsockname'
ld: lib/liblldbHost.a(TCPSocket.cpp.o): in function `lldb_private::TCPSocket::GetLocalIPAddress() const':
lldb/source/Host/common/TCPSocket.cpp:104: undefined reference to `getsockname'
ld: lib/liblldbHost.a(TCPSocket.cpp.o): in function `lldb_private::TCPSocket::GetRemotePortNumber() const':
lldb/source/Host/common/TCPSocket.cpp:114: undefined reference to `getpeername'
ld: lib/liblldbHost.a(TCPSocket.cpp.o): in function `lldb_private::TCPSocket::GetRemoteIPAddress() const':
lldb/source/Host/common/TCPSocket.cpp:126: undefined reference to `getpeername'
ld: lib/liblldbHost.a(TCPSocket.cpp.o): in function `lldb_private::TCPSocket::Connect(llvm::StringRef)':
lldb/source/Host/common/TCPSocket.cpp:172: undefined reference to `connect'
ld: lib/liblldbHost.a(TCPSocket.cpp.o): in function `lldb_private::TCPSocket::Listen(llvm::StringRef, int)':
lldb/source/Host/common/TCPSocket.cpp:213: undefined reference to `setsockopt'
ld: lldb/source/Host/common/TCPSocket.cpp:223: undefined reference to `bind'
ld: lldb/source/Host/common/TCPSocket.cpp:225: undefined reference to `listen'
ld: lldb/source/Host/common/TCPSocket.cpp:235: undefined reference to `getsockname'
ld: lib/liblldbHost.a(UDPSocket.cpp.o): in function `lldb_private::UDPSocket::Send(void const*, unsigned long)':
lldb/source/Host/common/UDPSocket.cpp:40: undefined reference to `sendto'
ld: lib/liblldbHost.a(UDPSocket.cpp.o): in function `lldb_private::UDPSocket::Connect(llvm::StringRef, bool)':
lldb/source/Host/common/UDPSocket.cpp:79: undefined reference to `getaddrinfo'
ld: lldb/source/Host/common/UDPSocket.cpp:82: undefined reference to `gai_strerror'
ld: lldb/source/Host/common/UDPSocket.cpp:106: undefined reference to `freeaddrinfo'
ld: lldb/source/Host/common/UDPSocket.cpp:126: undefined reference to `bind'
ld: lldb/source/Host/common/UDPSocket.cpp:130: undefined reference to `getsockname'
ld: lib/liblldbHost.a(DomainSocket.cpp.o): in function `lldb_private::DomainSocket::Connect(llvm::StringRef)':
lldb/source/Host/posix/DomainSocket.cpp:85: undefined reference to `connect'
ld: lib/liblldbHost.a(DomainSocket.cpp.o): in function `lldb_private::DomainSocket::Listen(llvm::StringRef, int)':
lldb/source/Host/posix/DomainSocket.cpp:104: undefined reference to `bind'
ld: lldb/source/Host/posix/DomainSocket.cpp:106: undefined reference to `listen'
ld: lib/liblldbHost.a(DomainSocket.cpp.o): in function `lldb_private::DomainSocket::GetSocketName() const':
lldb/source/Host/posix/DomainSocket.cpp:134: undefined reference to `getpeername'
ld: lib/liblldbHost.a(SocketAddress.cpp.o): in function `lldb_private::SocketAddress::GetIPAddress() const':
lldb/source/Host/common/SocketAddress.cpp:140: undefined reference to `inet_ntop'
ld: lldb/source/Host/common/SocketAddress.cpp:145: undefined reference to `inet_ntop'
ld: lib/liblldbHost.a(SocketAddress.cpp.o): in function `lldb_private::SocketAddress::GetAddressInfo(char const*, char const*, int, int, int, int)':
lldb/source/Host/common/SocketAddress.cpp:234: undefined reference to `getaddrinfo'
ld: lldb/source/Host/common/SocketAddress.cpp:243: undefined reference to `freeaddrinfo'
ld: lib/liblldbHost.a(SocketAddress.cpp.o): in function `lldb_private::SocketAddress::SetToLocalhost(unsigned char, unsigned short)':
lldb/source/Host/common/SocketAddress.cpp:260: undefined reference to `in6addr_loopback'
ld: lib/liblldbHost.a(SocketAddress.cpp.o): in function `lldb_private::SocketAddress::SetToAnyAddress(unsigned char, unsigned short)':
lldb/source/Host/common/SocketAddress.cpp:282: undefined reference to `in6addr_any'
ld: lib/liblldbHost.a(SocketAddress.cpp.o): in function `lldb_private::SocketAddress::IsAnyAddr() const':
lldb/source/Host/common/SocketAddress.cpp:294: undefined reference to `in6addr_any'
ld: lib/liblldbHost.a(SocketAddress.cpp.o): in function `lldb_private::SocketAddress::IsLocalhost() const':
lldb/source/Host/common/SocketAddress.cpp:300: undefined reference to `in6addr_loopback'
ld: lib/liblldbPluginProcessGDBRemote.a(ProcessGDBRemote.cpp.o): in function `lldb_private::process_gdb_remote::ProcessGDBRemote::LaunchAndConnectToDebugserver(lldb_private::ProcessInfo const&)':
lldb/source/Plugins/Process/gdb-remote/ProcessGDBRemote.cpp:3456: undefined reference to `socketpair'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Comment by kenz-gelsoft** [comment]
for Host.cpp
```
~/src/llvm-project/lldb/source/Host> find . -iname Host\*.cpp | grep -v HostInfo
./common/Host.cpp
./common/HostNativeThreadBase.cpp
./common/HostProcess.cpp
./common/HostThread.cpp
./freebsd/Host.cpp
./linux/Host.cpp
./netbsd/HostNetBSD.cpp
./openbsd/Host.cpp
./posix/HostProcessPosix.cpp
./posix/HostThreadPosix.cpp
./windows/Host.cpp
./windows/HostProcessWindows.cpp
./windows/HostThreadWindows.cpp
```

* Linux uses procfs
* BSDs uses syscall

**Comment by kenz-gelsoft** [comment]
POSIX socket API needs to be linked with network lib.

**Comment by kenz-gelsoft** [comment]
Hmm, this change didn't work
```diff
diff --git a/lldb/source/Host/CMakeLists.txt b/lldb/source/Host/CMakeLists.txt
index 08a647858..4739c892f 100644
--- a/lldb/source/Host/CMakeLists.txt
+++ b/lldb/source/Host/CMakeLists.txt
@@ -159,6 +159,10 @@ if (WIN32)
   list(APPEND LLDB_SYSTEM_LIBS psapi)
 endif()
 
+if (CMAKE_SYSTEM_NAME MATCHES "Haiku")
+  list(APPEND LLDB_SYSTEM_LIBS network)
+endif ()
+
 if (LLDB_ENABLE_LIBEDIT)
   list(APPEND LLDB_LIBEDIT_LIBS ${LibEdit_LIBRARIES})
   if (LLVM_BUILD_STATIC)
```

ld command line doesn't contain (lib)network

need to investigate how to build `-o lib/liblldb.so.12.0.1`

**Comment by kenz-gelsoft** [comment]
fixed in 1a774db...2e638a1
